packages/vite/src/node/server/transformRequest.ts

```diff
@@ -169,7 +172,62 @@ async function loadAndTransform(
 
   // load
   const loadStart = isDebug ? performance.now() : 0
-  const loadResult = await pluginContainer.load(id, { ssr })
+  let loadResult: LoadResult
+
+  const idWithoutHmrFlag = id.replace(DEP_VERSION_RE, '')
+  const loadCacheKey = ssr ? `ssr:${idWithoutHmrFlag}` : idWithoutHmrFlag
+
+  loadResult = await pluginContainer.load(id, { ssr })
+
+  const includedInPersistentCache =
+    _persistentCache &&
+    !file.includes(server.config.cacheDir) &&
+    !file.includes('vite/dist/client') &&
```

Comment:
this line could use a comment as to either when it would occur or why we want to exclude it

Reply:
Well it excludes the vite client

Reply:
> Well it excludes the vite client

That doesn't explain why 😆